### PR TITLE
AAE-11711: Improve slack notification payload

### PR DESCRIPTION
**Pull request by atchertchian**
- fix information displayed depending on event use cases (push, pull_request, schedule, workflow_dispatch)
- fix management of messages with quotes
- improve overall display

### CODE REVIEW

File: .github/actions/send-slack-notification/compute-message.sh [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
+        COMPUTED_MESSAGE=$COMMIT_MESSAGE
         ;;
     esac
 fi
 
-echo ''
-echo 'EOF'
+if [ -n "$COMPUTED_MESSAGE" ]; then
+    echo 'result<<EOF'
+    echo -n "*Message*\n${COMPUTED_MESSAGE}" | sed -z 's/\n/\\n/g' | sed -r "s/\\\"/\\\\\"/g"
```
Comment (atchertchian) [inline .github/actions/send-slack-notification/compute-message.sh:22]: codacy says
```
echo won't expand escape sequences. Consider printf.
```

File: .github/actions/send-slack-notification/compute-message.sh [hunk 2]
```diff
@@ -1,22 +1,27 @@
 #!/bin/bash -e
 
-echo 'result<<EOF'
-
+COMPUTED_MESSAGE=""
 if [ -n "$BLOCK_MESSAGE" ]; then
-    echo -n "${BLOCK_MESSAGE}" | sed -z 's/\n/\\n/g'
+    COMPUTED_MESSAGE=$BLOCK_MESSAGE
```
Comment (alxgomz) [inline .github/actions/send-slack-notification/compute-message.sh:5]:
```suggestion
    COMPUTED_MESSAGE="$BLOCK_MESSAGE"
```

File: .github/actions/send-slack-notification/compute-message.sh [hunk 3]
```diff
@@ -1,22 +1,27 @@
...
 else
     case $EVENT_NAME in
         pull_request)
-        echo -n "${PR_TITLE}" | sed -z 's/\n/\\n/g'
+        COMPUTED_MESSAGE=$PR_TITLE
```
Comment (alxgomz) [inline .github/actions/send-slack-notification/compute-message.sh:9]:
```suggestion
        COMPUTED_MESSAGE="$PR_TITLE"
```

File: .github/actions/send-slack-notification/compute-message.sh [hunk 4]
```diff
@@ -1,22 +1,27 @@
...
         ;;
         issues)
-        echo -n "${ISSUE_BODY}" | sed -z 's/\n/\\n/g'
+        COMPUTED_MESSAGE=$ISSUE_BODY
```
Comment (alxgomz) [inline .github/actions/send-slack-notification/compute-message.sh:12]:
```suggestion
        COMPUTED_MESSAGE="$ISSUE_BODY"
```

File: .github/actions/send-slack-notification/compute-message.sh [hunk 5]
```diff
@@ -1,22 +1,27 @@
...
         ;;
         *)
-        echo -n "${COMMIT_MESSAGE}" | sed -z 's/\n/\\n/g'
+        COMPUTED_MESSAGE=$COMMIT_MESSAGE
```
Comment (alxgomz) [inline .github/actions/send-slack-notification/compute-message.sh:15]:
```suggestion
        COMPUTED_MESSAGE="$COMMIT_MESSAGE"
```

**Review by alxgomz — COMMENTED** [review]
I think this bash vars are prone to errors as is

**Review by alxgomz — DISMISSED** [review]
💅

**Review by alxgomz — APPROVED** [review]

**Review by erdemedeiros — APPROVED** [review]
LGTM!

**Pull request merged**